Admin/AbstractAdmin.php

```diff
@@ -1531,6 +1532,23 @@ public function setPersistFilters($persist)
         );
 
         $this->persistFilters = $persist;
+
+        if ($persist === true) {
+            try {
+                $request = $this->getRequest();
+                $session = $request->getSession();
+
+                if ($session === null) {
+                    throw new \RuntimeException('The Request object has not session available');
+                }
+
+                $this->filterPersister = new SessionFilterPersister($session);
+            } catch (\RuntimeException $exception) {
+                // no request nor session
```

Comment:
Since the container is dumped after this method is called, that would create something useless so yeah, I think an exception should be thrown when this method is misused. Or at least the catch should contain a deprecation notice saying an exception will be thrown in the future, if you think this could break existing setups.